tests/e2e/e2e_util_test.go

```diff
@@ -106,3 +108,155 @@ func (s *IntegrationTestSuite) sendIBC(srcChainID, dstChainID, recipient string,
 
 	s.T().Log("successfully sent IBC tokens")
 }
+
+func (s *IntegrationTestSuite) submitProposal(c *chain.Chain) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("submitting upgrade proposal for chain-id: %s", c.ChainMeta.Id)
+	exec, err := s.dkrPool.Client.CreateExec(docker.CreateExecOptions{
+		Context:      ctx,
+		AttachStdout: true,
+		AttachStderr: true,
+		Container:    s.valResources[c.ChainMeta.Id][0].Container.ID,
+		User:         "root",
+		Cmd: []string{
+			"osmosisd", "tx", "gov", "submit-proposal", "software-upgrade", "v8", "--title=\"v8 upgrade\"", "--description=\"v8 upgrade proposal\"", "--upgrade-height=75", "--upgrade-info=\"\"", fmt.Sprintf("--chain-id=%s", c.ChainMeta.Id), "--from=val", "-b=block", "--yes", "--keyring-backend=test",
+		},
+	})
+	s.Require().NoError(err)
+
+	var (
+		outBuf bytes.Buffer
+		errBuf bytes.Buffer
+	)
+
+	err = s.dkrPool.Client.StartExec(exec.ID, docker.StartExecOptions{
+		Context:      ctx,
+		Detach:       false,
+		OutputStream: &outBuf,
+		ErrorStream:  &errBuf,
+	})
+
+	s.Require().NoErrorf(
```

Comment:
I think regardless of the outcome of the command this never error never gets triggered. My thought was to instead marshal the response and only unmarshal the error code and require code 0. Is this a good approach or is there a better way of doing this @p0mvn @alexanderbez

Reply:
I instead opted to just check the outBuf.String() for code 0, I think it works and is simpler.